src/canard.c

```diff
@@ -1,6 +1,736 @@
+/*
+ * Copyright (C) 2015 Michael Sierra <sierramichael.a@gmail.com>
+ */
+
 #include "canard.h"
 #include "canard_internals.h"
 
+#define CANARD_MAKE_TRANSFER_DESCRIPTOR(data_type_id, transfer_type, \
+                                        src_node_id, dst_node_id) \
+    ((data_type_id) | ((transfer_type) << 16) | \
+     ((src_node_id) << 18) | ((dst_node_id) << 25))
+
+struct CanardTxQueueItem
+{
+  CanardTxQueueItem* next;
+  CanardCANFrame frame;
+};
+
+/**
+ *  API functions
+ */
+
+/**
+ * Initializes the library state.
+ * Local node ID will be set to zero, i.e. the node will be anonymous.
+ */
+void canardInit(CanardInstance* out_ins, canardOnTransferReception on_reception)
+{
+  out_ins->node_id = CANARD_BROADCAST_NODE_ID;
+  out_ins->on_reception = on_reception;
+  out_ins->rx_states = NULL;
+  out_ins->tx_queue = NULL;
+  canardInitPoolAllocator(&out_ins->allocator, out_ins->buffer, CANARD_AVAILABLE_BLOCKS);
```

Comment:
I suggest to avoid hard-coding the pool size. Instead, we could make the initialization function accept a pointer to the buffer:

```c
void canardInit(CanardInstance* out_ins, CanardOnTransferReception on_reception,
                void* mem_arena, size_t mem_arena_size)
{
    ...
    canardInitPoolAllocator(&out_ins->allocator, mem_arena, mem_arena_size / CANARD_MEM_BLOCK_SIZE);
}
```